modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,246 @@
+
+		String sourceLanguageCode = _getLanguageCode(slc);
+
+		Map<String, String> translatedFieldsMap = new HashMap<>();
+		Map<String, String> fieldsMap = translatorPacket.getFieldsMap();
+
+		for (Map.Entry<String, String> entry : fieldsMap.entrySet()) {
+			translatedFieldsMap.put(
+				entry.getKey(),
+				_translate(
+					deepLTranslatorConfiguration.url(),
+					deepLTranslatorConfiguration.authKey(), entry.getValue(),
+					sourceLanguageCode, targetLanguageCode));
+		}
+
+		return new TranslatorPacket() {
+
+			@Override
+			public long getCompanyId() {
+				return translatorPacket.getCompanyId();
+			}
+
+			@Override
+			public Map<String, String> getFieldsMap() {
+				return translatedFieldsMap;
+			}
+
+			@Override
+			public String getSourceLanguageId() {
+				return translatorPacket.getSourceLanguageId();
+			}
+
+			@Override
+			public String getTargetLanguageId() {
+				return translatorPacket.getTargetLanguageId();
+			}
+
+		};
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_deepLTranslatorConfiguration = ConfigurableUtil.createConfigurable(
+			DeepLTranslatorConfiguration.class, properties);
+	}
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		String ret = list.get(0);
+
+		return StringUtil.toUpperCase(ret);
+	}
+
+	private List<String> _getSupportedLanguages(
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration) {
+
+		List<SupportedLanguage> supportedLanguages = null;
+
+		try {
+			supportedLanguages = _deepLClient.verifySupportedLanguage(
+				deepLTranslatorConfiguration.authKey(), DeepLConstants.TARGET,
+				DeepLConstants.SUPPORTED_LANGUAGE_INQ_URL);
+		}
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return new ArrayList<>();
+		}
+		catch (TranslatorException translatorException) {
+			throw new RuntimeException(translatorException);
+		}
+
+		List<String> languages = new ArrayList<>();
+
+		supportedLanguages.forEach(sl -> languages.add(sl.getLanguage()));
+
+		return languages;
+	}
+
+	private String _translate(
+			String url, String authKey, String text, String sourceLanguageCode,
+			String targetLanguageCode)
+		throws TranslatorException {
+
+		if (Validator.isBlank(text)) {
+			return text;
+		}
+
+		TranslateResponse translateResponse = null;
+
+		try {
+			translateResponse = _deepLClient.execute(
+				authKey, text, sourceLanguageCode, targetLanguageCode, url);
+		}
+		catch (IOException ioException) {
+			throw new TranslatorException(
+				"DeepL translator returns original text. " +
+					ioException.getLocalizedMessage());
+		}
+
+		List<Translation> translations = translateResponse.getTranslations();
+
+		Translation translation = translations.get(0);
+
+		return translation.getText();
```

Comment:
Same as before. Can you move all the code in the happy path inside the try section?